src/test/regress/expected/shard_rebalancer.out

```diff
@@ -2378,3 +2378,63 @@ SELECT rebalance_table_shards();
 
 CALL citus_cleanup_orphaned_shards();
 DROP TABLE test_rebalance_with_index CASCADE;
+-- Test rebalancer with disabled worker
+SET citus.shard_replication_factor TO 2;
+DROP TABLE IF EXISTS test_rebalance_with_disabled_worker;
+CREATE TABLE test_rebalance_with_disabled_worker (a int);
+SELECT create_distributed_table('test_rebalance_with_disabled_worker', 'a', colocate_with:='none');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT citus_disable_node('localhost', :worker_2_port);
+ citus_disable_node
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT public.wait_until_metadata_sync(30000);
+ wait_until_metadata_sync
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT rebalance_table_shards('test_rebalance_with_disabled_worker');
+ rebalance_table_shards
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT citus_activate_node('localhost', :worker_2_port);
+ citus_activate_node
+---------------------------------------------------------------------
+                  35
+(1 row)
+
+DROP TABLE test_rebalance_with_disabled_worker;
+-- Test rebalance with all shards excluded
+DROP TABLE IF EXISTS test_with_all_shards_excluded;
+CREATE TABLE test_with_all_shards_excluded(a int);
```

Comment:
minor: maybe use PRIMARY KEY in the table such that on enterprise repo, the test is triggered  via non-blocking rebalancer such that we get extra coverage. 
(I know the code-path is the same until we get to the actual rebalance -- and we don't to rebalance for this test--, but still it might change in the future.)